File: 0219. Contains Duplicate II.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        if(nums.Length <= 1){
            return false;
        } 

        
        Dictionary<int, int> map = new Dictionary<int, int>();

        for(int i = 0;i<nums.Length;i++){
            var v = nums[i];
            if(map.ContainsKey(v) && (i - map[v] <= k)){
                return true;
            }
            map[v] = i;
        }

        
        return false;
    }
}

File: 0219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if len(nums) <= 1:
            return False
         

        map = {}
        
        for i in range(len(nums)):
            v = nums[i]
            if map.__contains__(v) and (i - map[v] <= k):
                return True
            map[v] = i
        
        return False